packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -747,8 +747,8 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   // Calculate the new scroll offset so the cursor remains visible.
   double _getScrollOffsetForCaret(Rect caretRect) {
-    final double caretStart = _isMultiline ? caretRect.top : caretRect.left;
-    final double caretEnd = _isMultiline ? caretRect.bottom : caretRect.right;
+    final double caretStart = _isMultiline ? caretRect.top - kCaretHeightOffset : caretRect.left;
+    final double caretEnd = _isMultiline ? caretRect.bottom + kCaretHeightOffset : caretRect.right;
```

Comment:
When we were deciding where to scroll to, we scrolled so that the cursor was visible.  This means that a few pixels of line height were not visible on the top/bottom, causing the bug.  This fix adds in the offset so that the full line is visible, not just the visible part of the cursor.

@HansMuller What do you think of this solution?  It looks like actually Gary's Strut changes won't fix this.  If something like this seems like a valid idea then I'll write some tests.  On my machine this change doesn't break any existing unit tests.